pkg/operator/ceph/cluster/mgr/dashboard.go

```diff
@@ -121,27 +121,36 @@ func (c *Cluster) configureDashboardModule(m *mgrConfig) error {
 		return err
 	}
 
-	// server port
-	port := strconv.Itoa(m.DashboardPort)
-	changed, err := client.MgrSetConfig(c.context, c.Namespace, m.DaemonID, c.clusterInfo.CephVersion, "mgr/dashboard/server_port", port, false)
-	if err != nil {
-		return err
-	}
-	hasChanged = hasChanged || changed
-
 	// ssl support
 	var ssl string
 	if c.dashboard.SSL == nil {
 		ssl = ""
 	} else {
 		ssl = strconv.FormatBool(*c.dashboard.SSL)
 	}
-	changed, err = client.MgrSetConfig(c.context, c.Namespace, m.DaemonID, c.clusterInfo.CephVersion, "mgr/dashboard/ssl", ssl, false)
+	changed, err := client.MgrSetConfig(c.context, c.Namespace, m.DaemonID, c.clusterInfo.CephVersion, "mgr/dashboard/ssl", ssl, false)
 	if err != nil {
 		return err
 	}
 	hasChanged = hasChanged || changed
 
+	// server port
+	port := strconv.Itoa(m.DashboardPort)
+	changed, err = client.MgrSetConfig(c.context, c.Namespace, m.DaemonID, c.clusterInfo.CephVersion, "mgr/dashboard/server_port", port, false)
+	if err != nil {
+		return err
+	}
+	hasChanged = hasChanged || changed
+
+	// SSL enabled. Needed to set specifically the ssl port setting starting with Nautilus.
+	if ssl == "true" {
+		changed, err = client.MgrSetConfig(c.context, c.Namespace, m.DaemonID, c.clusterInfo.CephVersion, "mgr/dashboard/ssl_server_port", port, false)
```

Comment:
does the version of ceph running also need to be checked here?

Reply:
Yes, that is one possibility! But in this way is more easy and robust.
I think that in this case, the situation where the problem is found is only when the dashboard is enabled with SSL in Nautilus, and i estimate that this use case is going to be almost non existent. 
Another question related with this is ... what would happen if the installation use a Nautilus version prior to the introduction of "ssl_server_port"?  (this is even less probable , maybe a crazy developer doing tests ...), but it can happen...
So doing the minimum to solve the problem in all the possible use cases seems to be the right approach.
Maybe the error message can be improved. What do you think?

Reply:
I haven't thought through all of the combinations, but there is also the possibility of mimic or users that upgrade rook but don't upgrade ceph at the same time. if you grep around for CephVersion there is a library in Rook that makes it really easy to check for the ceph version being run.

Reply:
@noahdesu : I didn't know the library. Thanks for pointing this!. I have change the implementation in order to check the Ceph version.

The code including the new ssl setting was merged in Nautilus commit #e3f2ede (2019-04-05). 
Therefore 14.2.1 (2019-04-25) is the first Ceph version where the new ssl port setting is present.

The only problem i see with this approach is that if somebody tries to use this setting with a copy of Ceph between 2019-04-05 and 2019-04-25 it is not going to work despite the port ssl setting is already present. 
In any case, I think that this a very improbable situation, and the solution is better/more clean using the CephVersion library. Thanks for your help!.